src/gulp-jsdoc.js

```diff
@@ -81,8 +89,6 @@ export function jsdoc(config, done) {
                     args = args.concat(['-t', inkdocstrap]);
                 }
 
-                args = args.concat(files);
```

Comment:
A simple test: change in the source dir the config file and run doc:
```
    "excludePattern": ".+\\.js(doc)?$"
```
You will see that jsdoc complains because it applies these filters to the input files. This case I would imagine is common - glob everything and filter in the config. 

Given this, it might be best to always use source, but in a slightly different way than you suggest. Instead of setting it, simply check if source.include is there, if so append; if not add the files. That would appear to be safe (I'll move call this a breakng change and move to 1.0 for this anyway). See: http://usejsdoc.org/about-configuring-jsdoc.html#specifying-input-files
Something like:
```jsdocConfig.source = _merge(jsdocConfig.source || {}, {include: [files]});```